clients/vault/src/system.rs

```diff
@@ -860,11 +888,7 @@ impl VaultService {
 				)
 				.await
 				.map_err(|e| Error::RuntimeError(e))
-		} else if let Some(_faucet_url) = &self.config.faucet_url {
-			tracing::info!("[{}] Automatically registering...", vault_id.pretty_print());
-			// TODO
-			// faucet::fund_and_register(&self.spacewalk_parachain, faucet_url, &vault_id)
-			// 	.await?;
+		} else if self.try_fund_from_faucet().await {
```

Comment:
here we want to know if it was successful or not; if it was, then all good. If not, an error is returned.